docs/USAGE_GUIDE.md

```diff
@@ -0,0 +1,529 @@
+2. Properties (or aka Java Fields)
+   1. Exactly the same fundamentals as the Page Object. These fields are the elements that exist on the page. With a "non-list" component, we always want to provide the fully qualified By locator that includes the parent.
+3. Getters
+   1. Standard getter functions are auto generated with the lombok `@Getter` annotation from the class level. Any time we want to interact with an element, we
+      always call the getter instead of the raw field.
+4. No Constructor
+   1. Here, we don't need to customize our constructor with a verification method. The Page Object is responsible for this.
+5. Helper Functions
+   1. We can include any additional helper functions here to make our calling code easier to read.
+6. Instantiating the Component on the Page Object
+   1. The Page Object can then instantiate a new instance of the Component we just wrote. Code calling the loginPage will now have access to getHeaderComponent()'s getters and helper functions!
+
+##### Component Example - List Context
+Building a Component under a "list" context is fundamentally the same. In the search results page example mentioned from the Components section, we
+can define properties exactly the same way. The only difference is that a list Component doesn't define properties with the parent in the `By` locator.
+The search results page would then create the list of the search results with `BaseComponent`'s `buildComponentList()` method. 
+
+**The Component**
+```java
+@Getter
+public class SearchResultItem extends BaseComponent {
+
+   private final DivWebElement itemName = new DivWebElement(".inventory_item_name");
+   private final DivWebElement itemDescription = new DivWebElement(".inventory_item_desc");
+   private final DivWebElement itemPrice = new DivWebElement(".inventory_item_price");
+   private final ButtonWebElement addToCart = new ButtonWebElement(".btn_primary");
+
+   public void clickAddToCartButton() {
+      getAddToCart().click();
+   }
+}
+```
+
+**The Page Object**
+```java
+@Getter
+public class SearchResultsPage extends BasePage {
+   private final static String INVENTORY_ITEM_SELECTOR = ".inventory_item";
+   private final HeaderComponent headerComponent = new HeaderComponent();
+   private final DivWebElement inventoryListContainer = new DivWebElement(By.cssSelector(".inventory_list"));
+   private final DropDownWebElement sortDropDown = new DropDownWebElement(".product_sort_container");
+
+   public SearchResultsPage() {
+      verifyIsOnPage(getInventoryList());
+   }
+
+   public List<SearchResultItem> getSearchResultsList() {
+      var listOfElements = getInventoryListContainer().findElements(DivWebElement.class, INVENTORY_ITEM_SELECTOR);
+      return buildComponentList(listOfElements, SearchResultItem.class);
+   }
```

Comment:
Can you add a note in the breakdown below highlighting why you should not make the `buildComponentList()` method in a private final declaration?

This would force a lookup when this variable is made which is completely against the idea of scaffold 🍡 

I'm fairly certain someone will try to do that, i.e. do the following:

```java
private final DivWebElement container = new DivWebElement(".container");
private final List<Component> componentList = buildComponentList(getContainer().findElements("blah"), Component.class);

```

Reply:
The final modifier here actually doesn't break usage. Any time we interact with an element, the element is always searched for first and we don't use old references. 

This example looks like usage on a class file, right? I think this is still okay, although centralizing it to a Page Object would be better.

Reply:
note: circle back on this, the example intended is instantiation as a class variable

Reply:
Great comment on this, I misunderstood at first. I updated the documentation for this on the next commit by adding in a "best practices" subsection for page objects and components.